homeassistant/components/tuya/select.py

```diff
@@ -219,6 +220,30 @@
             entity_category=EntityCategory.CONFIG,
         ),
     ),
+    # Robot Vacuum
+    # https://developer.tuya.com/en/docs/iot/fsd?id=K9gf487ck1tlo
+    "sd": (
+        SelectEntityDescription(
+            key=DPCode.CISTERN,
+            name="Water Tank Adjustment",
+            entity_category=EntityCategory.CONFIG,
+            device_class=DEVICE_CLASS_HUMIDITY,
+            icon="mdi:water-opacity",
+        ),
+        SelectEntityDescription(
+            key=DPCode.COLLECTION_MODE,
+            name="Dust Collection Mode",
+            entity_category=EntityCategory.CONFIG,
+            device_class=DEVICE_CLASS_PRESSURE,
+            icon="mdi:air-filter",
+        ),
+        SelectEntityDescription(
+            key=DPCode.MODE,
```

Comment:
Select Mode is not part of the vacuum entity.

The device has the ability to choose several modes not foreseen in the vacuum entity, this justifies the inclusion of the select Mode.

![image](https://user-images.githubusercontent.com/31328123/144585275-f822f638-574e-4f9c-a95f-a159e65fb2cc.png)